daemon/health.go

```diff
@@ -148,12 +148,20 @@ func handleProbeResult(d *Daemon, c *container.Container, result *types.Healthch
 		// then we check if we are within the start period of the container in which
 		// case we do not increment the failure streak.
 		if h.Status == types.Starting {
-			startPeriod := timeoutWithDefault(c.Config.Healthcheck.StartPeriod, defaultStartPeriod)
-			timeSinceStart := result.Start.Sub(c.State.StartedAt)
+			if result.Start.IsZero() {
+				// In case this is the first healthcheck, the failing streak
+				// will only be incremented when startPeriod < 0
+				if c.Config.Healthcheck.StartPeriod >= 0 {
```

Comment:
Perhaps it would be clearer to introduce a new boolean flag for this, instead of overloading `StartPeriod` with a magic value.